### PR TITLE
Update free transfer logic

### PR DESCRIPTION
**Pull request by bfauble**
#### Summary of changes
**Asana Ticket:** [Make some additional transfers free in fare calculation](https://app.asana.com/0/555089885850811/1202313545967339/f)

update transfer calculations to accommodate:
express_bus -> express_bus
bus -> bus -> subway
subway -> bus -> bus

image
image

### CODE REVIEW

**Review by thecristen — CHANGES_REQUESTED** [review]
Some minor comments. Smooth! 🏄🏼

File: apps/trip_plan/lib/trip_plan/transfer.ex [hunk 1]
```diff
@@ -32,9 +37,23 @@ defmodule TripPlan.Transfer do
   def is_subway_transfer?([_ | legs]), do: is_subway_transfer?(legs)
 
   def is_subway_transfer?(_), do: false
+  require Logger
 
   @doc "Takes a pair of legs and returns true if there might be a transfer between the two, based on the list in @single_ride_transfers. Exception: no transfers from bus route to same bus route."
```
[the remark this answers is not in the document]
Comment (bfauble) [reply apps/trip_plan/lib/trip_plan/transfer.ex:42]: bah, super long line was hidden off the screen, good catch

File: apps/trip_plan/lib/trip_plan/transfer.ex [hunk 2]
```diff
@@ -32,9 +37,23 @@ defmodule TripPlan.Transfer do
   def is_subway_transfer?([_ | legs]), do: is_subway_transfer?(legs)
 
   def is_subway_transfer?(_), do: false
+  require Logger
```
[the remark this answers is not in the document]
Comment (bfauble) [reply apps/trip_plan/lib/trip_plan/transfer.ex:40]: was just for some debugging, removing

File: apps/trip_plan/lib/trip_plan/transfer.ex [hunk 3]
```diff
@@ -32,9 +37,23 @@ defmodule TripPlan.Transfer do
...
   @spec is_maybe_transfer?([Leg.t()]) :: boolean
+  def is_maybe_transfer?([
+        first_leg = %Leg{mode: %TransitDetail{route_id: first_route}},
+        middle_leg = %Leg{mode: %TransitDetail{route_id: middle_route}},
+        last_leg = %Leg{mode: %TransitDetail{route_id: last_route}}
+      ]) do
+    @single_ride_transfers
+    |> Map.get(:triple)
+    |> Map.get(Fares.to_fare_atom(first_route), [])
+    |> Kernel.==(Enum.map([first_route, middle_route, last_route], &Fares.to_fare_atom/1))
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/trip_plan/lib/trip_plan/transfer.ex:50]: I feel like it is free.. but I can't claim that unambiguously. But I'm ok leaving this as it is - we already cover ourselves with our "may be less with transfers" disclaimer, and can always fix it later if it's a definite issue.

**Review by thecristen — APPROVED** [review]
Thank you!!

**Pull request merged**